test/integration/resourcemanager/managedresourcecontroller/managedresource_controller_test.go

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package managedresourcecontroller_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+)
+
+const (
+	secretName           = "test-secret"
+	configMapName        = "test-configmap"
+	updateTestSecretName = "update-test-secret"
+)
+
+var _ = Describe("ManagedResource controller tests", func() {
+	var (
+		ctx              = context.Background()
+		secretForRef     *corev1.Secret
+		updateTestSecret *corev1.Secret
+		managedResource  *resourcesv1alpha1.ManagedResource
+		configMap        *corev1.ConfigMap
+	)
+	Context("successfull operations", func() {
+		BeforeEach(func() {
+			configMap = &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: corev1.SchemeGroupVersion.String(),
+					Kind:       "ConfigMap",
+				}, ObjectMeta: metav1.ObjectMeta{
+					Name:      configMapName,
+					Namespace: namespaceName,
+				}}
+
+			data, err := createSecretDataFromObject(configMap, "config-map.yaml")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(data).ToNot(BeNil())
+			secretForRef = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      secretName,
+					Namespace: namespaceName,
+				},
+				Data: *data,
+			}
+			Expect(testClient.Create(ctx, secretForRef)).To(Or(Succeed(), BeAlreadyExistsError()))
+
+			managedResource = &resourcesv1alpha1.ManagedResource{
```

Comment:
Nit: you could check if the `pkg/utils/managedresources/builder` pkg can be useful for creating the MR and its secret.

Reply:
Editing the resources directly gives us a bit more freedom for the tests, so I've avoided using the `pkg/utils/managedresources/builder` package.